trust-quorum/src/coordinator_state.rs

```diff
@@ -0,0 +1,195 @@
+    ///
+    /// Return the newly constructed `CoordinatorState` along with this node's
+    /// `PrepareMsg` so that it can be persisted.
+    ///
+    /// Precondition: This node must be a member of the new configuration
+    /// or this method will panic. This is ensured as part of passing in a
+    /// `ValidatedReconfigureMsg`.
+    pub fn new_uninitialized(
+        my_platform_id: PlatformId,
+        now: Instant,
+        msg: ValidatedReconfigureMsg,
+    ) -> Result<(CoordinatorState, PrepareMsg), ReconfigurationError> {
+        // Create a configuration for this epoch
+        let (config, shares) =
+            Configuration::new(my_platform_id.clone(), &msg)?;
+
+        let shares_by_member: BTreeMap<PlatformId, Share> = config
+            .members
+            .keys()
+            .cloned()
+            .zip(shares.shares.expose_secret().iter().cloned())
+            .collect();
+
+        let mut prepares = BTreeMap::new();
+        let mut my_prepare_msg: Option<PrepareMsg> = None;
+        for (platform_id, share) in shares_by_member.into_iter() {
+            let prepare_msg = PrepareMsg { config: config.clone(), share };
+            if platform_id == my_platform_id {
+                // The prepare message to add to our `PersistentState`
+                my_prepare_msg = Some(prepare_msg);
+            } else {
+                // Create a message that requires sending
+                prepares.insert(platform_id, prepare_msg);
+            }
+        }
+        let op = CoordinatorOperation::Prepare {
+            prepares,
+            prepare_acks: BTreeSet::new(),
+        };
+
+        let state = CoordinatorState::new(my_platform_id, now, msg, config, op);
+        Ok((state, my_prepare_msg.unwrap()))
+    }
+
+    /// A reconfiguration from one group to another
+    pub fn new_reconfiguration(
+        my_platform_id: PlatformId,
+        now: Instant,
+        msg: ValidatedReconfigureMsg,
+        last_committed_config: &Configuration,
+    ) -> Result<CoordinatorState, ReconfigurationError> {
+        let (config, new_shares) =
+            Configuration::new(my_platform_id.clone(), &msg)?;
+
+        // We must collect shares from the last configuration
+        // so we can recompute the old rack secret.
+        let op = CoordinatorOperation::CollectShares {
+            epoch: last_committed_config.epoch,
+            members: last_committed_config.members.clone(),
+            collected_shares: BTreeMap::new(),
+            new_shares,
+        };
+
+        Ok(CoordinatorState::new(my_platform_id, now, msg, config, op))
+    }
+
+    // Intentionallly private!
+    fn new(
+        platform_id: PlatformId,
+        now: Instant,
+        reconfigure_msg: ValidatedReconfigureMsg,
+        configuration: Configuration,
+        op: CoordinatorOperation,
+    ) -> CoordinatorState {
+        // We always set the retry deadline to `now` so that we will send
```

Comment:
This isn't actually necessary anymore as messages get sent upon construction of a `CoordinatorState` by the `Node`. We should instead set the deadline to `now + reconfigure_msg.retry_timeout`.